akka-docs/rst/java/code/docs/http/javadsl/server/directives/CustomHttpMethodExamplesTest.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (C) 2015-2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package docs.http.javadsl.server.directives;
+
+import akka.NotUsed;
+import akka.actor.ActorSystem;
+import akka.event.LoggingAdapter;
+import akka.event.NoLogging$;
+import akka.http.javadsl.ConnectHttp;
+import akka.http.javadsl.Http;
+import akka.http.javadsl.ServerBinding;
+import akka.http.javadsl.model.*;
+import akka.http.javadsl.server.Route;
+import akka.http.javadsl.settings.ParserSettings;
+import akka.http.javadsl.settings.ServerSettings;
+import akka.http.javadsl.testkit.JUnitRouteTest;
+import akka.http.scaladsl.model.RequestEntityAcceptance;
+import akka.stream.Materializer;
+import akka.stream.javadsl.Flow;
+import org.junit.Test;
+
+import java.util.concurrent.CompletionStage;
+import java.util.concurrent.ExecutionException;
+
+import static akka.http.javadsl.model.HttpProtocols.HTTP_1_0;
+
+public class CustomHttpMethodExamplesTest extends JUnitRouteTest {
+
+  @Test
+  public void testComposition() throws InterruptedException, ExecutionException {
+    ActorSystem  system = system();
+    Materializer materializer = materializer();
+    LoggingAdapter loggingAdapter = NoLogging$.MODULE$;
+
+    int    port = 9090;
+    String host = "127.0.0.1";
+
+    //#customHttpMethod
+    HttpMethod BOLT =
+      HttpMethods.custom("BOLT", false, true, RequestEntityAcceptance.Expected$.MODULE$);
```

Comment:
Yah, agreed. Adding is a bit tricky because bin compat; HttpMethod must gain a `    public abstract RequestEntityAcceptance getRequestEntityAcceptance();` as we can't change return type of `    requestEntityAcceptance();
`